src/main/java/de/tum/cit/aet/artemis/lecture/domain/Lecture.java

```diff
@@ -136,17 +141,113 @@ public void setAttachments(Set<Attachment> attachments) {
         this.attachments = attachments;
     }
 
+    /**
+     * Get an unmodifiable list of lecture units when the objects are initialized by Hibernate.
+     * This is important so that external code does not modify the list without updating the back-references and order.
+     *
+     * @return the lecture units
+     */
```

Comment:
I think we could point out how to properly update the lecture units, as otherwise inexperienced devs who maybe do not grasp the idea might be confused by the immutable lecture units and just create another method that does not return a mutable object

```suggestion
    /**
     * Get an unmodifiable list of lecture units when the objects are initialized by Hibernate.
     * This is important so that external code does not modify the list without updating the back-references and order.
     * <p>
     * Use {@link #addLectureUnit}, {@link #removeLectureUnit}, {@link #setLectureUnits}, or {@link #reorderLectureUnits} to modify the lecture units.
     *
     * @return the lecture units
     */
```